amazonia/schema.yaml

```diff
@@ -0,0 +1,309 @@
+# In the default the yaml should have a flat structure like below (unless you always want an autoscaling unit or a
+# database unit to be created:
+#
+# unit_title: 'app2'
+# image_id: 'ami-1234some'
+# instance_type: 't2.nano'
+#
+# If the application specific yaml did not specify a unit title, image id, or instance type, amazonia would fill those
+# fields with either app2, ami-1234some, or t2.nano from the defaults depending on what was missing from the application yaml.
+#
+# See following list for values that can be provided.
+#
+# stack_key_list = ['stack_title',
+#                   'code_deploy_service_role',
+#                   'keypair',
+#                   'availability_zones',
+#                   'vpc_cidr',
+#                   'public_cidr',
+#                   'jump_image_id',
+#                   'jump_instance_type',
+#                   'nat_image_id',
+#                   'nat_instance_type',
+#                   'home_cidrs',
+#                   'autoscaling_units',
+#                   'database_units']
+#
+# unit_key_list = {'autoscaling_units': ['unit_title',
+#                                        'hosted_zone_name',
+#                                        'userdata',
+#                                        'image_id',
+#                                        'instance_type',
+#                                        'path2ping',
+#                                        'protocols',
+#                                        'loadbalancerports',
+#                                        'instanceports',
+#                                        'minsize',
+#                                        'maxsize',
+#                                        'health_check_grace_period',
+#                                        'iam_instance_profile_arn',
+#                                        'sns_topic_arn',
+#                                        'sns_notification_types',
+#                                        'elb_log_bucket',
+#                                        'health_check_type',
+#                                        'dependencies'],
+#                     'database_units': ['unit_title',
+#                                        'db_instance_type',
+#                                        'db_engine',
+#                                        'db_port']
+#                     }
+#
+
+
+unit_title: # A string containing the title to use for autoscaling or database units in your stack
+  type: 'string' # If unit title is not provided here, it must be provided as an embeded key in any autoscaling
+  nullable: True # or database units (if they are not also empty).
+  regex: '^([a-zA-Z0-9]+$)' # matches an alphanumeric title
+  empty: False
+image_id: # A string specifying which ami to use for autoscaling groups by its id
+  type: 'string'
+  empty: False
+  regex: '^([a][m][i][\-][a-zA-Z0-9]+$)' # matches an ami number eg: ami-some1234
```

Comment:
Should we use this regex? ^(ami-[a-zA-Z0-9]+$) for ami ids. Its the same but a bit cleaner